chatify/main.py

```diff
@@ -32,23 +34,37 @@ def __init__(self, shell=None, **kwargs):
         try:
             # If we find the config file we assume that user is a dev
             self.cfg = yaml.load(open("./config.yaml"), Loader=yaml.SafeLoader)
-            print("config.yaml file found!")
+            print("config.yaml file found; using custom configuration.")
             # Check dev config file
             check_dev_config(self.cfg)
 
-        except FileNotFoundError:
+        except FileNotFoundError: 
             # If we don't find the user is an end-point user
             dirname = pathlib.Path(__file__).parent.resolve()
             self.cfg = yaml.load(
                 open(pathlib.Path(str(dirname) + "/default_config.yaml")),
                 Loader=yaml.SafeLoader,
             )
-            print("Using default configuration!")
 
         self.prompts_config = self.cfg["prompts_config"]
         self.llm_chain = CreateLLMChain(self.cfg)
         self.tabs = None
 
+        # download local model if needed
```

Comment:
@jeremymanning, I am not sure why we are deleting the LLM model? If we are deleting it which LLM model is used?

Reply:
it's just so the LLM instance isn't held in memory when the model is first loaded.  depending on which model is used, it could take up quite a large amount of RAM.  i included the "del llm" line so that the model is removed from memory (to free up RAM) until the %%explain magic command is used again.  but when it's called next, even though the model still has to take up lots of RAM (again, only for certain models), the model weights and other stuff that takes a long time to download will have already been made available from that first call.

in other words, the "point" of initializing the LLM when the chatify extension is first registered is to trigger the download of the model weights from the very start, instead of causing a long delay when the magic command is called for the first time.

Reply:
from playing around with it a bit, i felt like this tweak improved the user experience.  but let me know what you think!

Reply:
Got it!

On another note, the [CreateLLMChain](https://github.com/ContextLab/chatify/blob/93b14c88785ffaf5c990d7e70486f770f7e80985/chatify/main.py#L49C1-L50C1) already performs an initialization of the LLM model. So we can skip the init part in the `main.py` function.